fedimint-client/src/module/mod.rs

```diff
@@ -581,6 +581,16 @@ pub trait ClientModule: Debug + MaybeSend + MaybeSync + 'static {
         false
     }
 
+    /// Gives module a chance to consolidate some inputs, e.g. when accumulating
+    /// a lot of notes of the same denomination.
+    async fn consolidate_inputs(
+        &self,
+        _dbtx: &mut DatabaseTransaction<'_>,
+        _operation_id: OperationId,
+    ) -> anyhow::Result<Vec<ClientInput<<Self::Common as ModuleCommon>::Input, Self::States>>> {
+        Ok(vec![])
+    }
+
```

Comment:
That's a  very e-cash specific API extension. Couldn't this just be part of `create_sufficient_input`? That one doesn't guarantee any particular amount anyway, it could be arbitrarily larger.

Reply:
Initially I was a bit afraid where should be added and if it will not break anything, so I made it a separate function. But now I think it will be OK as a part of `create_sufficient_input`.

Reply:
@elsirion OK, so I attempted it and one imediately difference is that now consolidations would only happen if the tx being built in under-funded, because only then we call `create_sufficient_input`.

I think it's OK, but worth noting if case I'm wrong.